tools/link.py

```diff
@@ -2111,6 +2118,27 @@ def create_worker_file(input_file, target_dir, output_file, options):
   utils.convert_line_endings_in_file(output_file, options.output_eol)
 
 
+def create_esm_wrapper(wrapper_file, support_target, wasm_target):
+  wasm_exports = ', '.join(shared.demangle_c_symbol_name(f) for f in settings.EXPORTED_FUNCTIONS)
+  wrapper = []
+  if wasm_exports:
+    wrapper.append(f"export {{ {wasm_exports} }} from './{settings.WASM_BINARY_FILE}';")
+  support_url = f'./{os.path.basename(support_target)}'
+  wrapper.append(f"import init from '{support_url}';")
+  wrapper.append('export default init;')
+  write_file(wrapper_file, '\n'.join(wrapper) + '\n')
+
+  # FIXME(sbc): This is a huge hack to rename the imports in the
+  # wasm file.  Find a better way to do this.
```

Comment:
We can add a binaryen pass for this in time, I suppose.

Reply:
Yes, I guess binaryen would be the place to do this.. although is it possible to do this without parsing and validating and re-writing the whole file?

Reply:
Ah yes, it would parse... I guess a binary-level rewrite could be more efficient, for compiles that never use binaryen at all.